contracts/staking/StakingPool.sol

```diff
@@ -0,0 +1,517 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+import "@openzeppelin/contracts-upgradeable/security/PausableUpgradeable.sol";
+
+import "../RoleAccessControl.sol";
+import "./StakingPoolLib.sol";
+
+contract StakingPool is
+    Initializable,
+    RoleAccessControl,
+    ReentrancyGuard,
+    PausableUpgradeable
+{
+    struct User {
+        uint128 depositAmount;
+        uint128[5] rewardAmounts;
+    }
+    struct RewardDue {
+        address token;
+        uint128 amount;
+    }
+
+    mapping(address => User) private _users;
+
+    uint32 private _rewardsAvailableTimestamp;
```

Comment:
@CjHare guessing we want individual getters for each of these!

Reply:
👍